### PR TITLE
First attempt at PySide2 support

### PR DESCRIPTION
**Pull request by hmaarrfk**
You can build pyside2 yourself or use
https://github.com/fredrikaverpil/pyside2-linux

I mostly searched for PySide and PyQt5 and kinda merged them together.

it ran the `examples/basics/scene/grid.py`, `magnify.py` and `cube.py`. Good enough for a first day. Haven't tested other things yet. But I think it is OK :D.

I had to add a `qWait` function.

I have no idea what `has_uic` is

### CODE REVIEW

File: vispy/util/tests/test_import.py [hunk 1]
```diff
@@ -80,7 +80,9 @@ def test_import_vispy_app2():
     """ Importing vispy.app should not pull in any backend toolkit. """
     allmodnames = loaded_vispy_modules('vispy.app', 2, True)
     assert_not_in('PySide', allmodnames)
+    assert_not_in('PySide2', allmodnames)
     assert_not_in('PyQt4', allmodnames)
+    # assert_not_in('PyQt5', allmodnames) # should PyQt5 be tested against too
```
Comment (davidh-ssec) [inline vispy/util/tests/test_import.py]: Yes probably.

File: vispy/app/qt.py [hunk 2]
```diff
@@ -17,11 +17,16 @@
 
 
 if qt_lib in ('pyqt4', 'pyside'):
+    # Why is this not a seperate option for pyside.
+    # PySide and PyQt4 are different aren't they?
```
Comment (davidh-ssec) [inline vispy/app/qt.py]: Yes this does seem strange doesn't it. If @larsoner or @kmuehlbauer don't have any ideas why this is like this, it seems like it should be split in to a separate if branch.

Comment (kmuehlbauer) [reply vispy/app/qt.py]: No idea, seems odd to me too.

Comment (larsoner) [reply vispy/app/qt.py]: Agreed this looks like a bug

File: vispy/app/backends/_qt.py [hunk 3]
```diff
@@ -143,7 +149,7 @@ def message_handler(*args):
 
     if qt_lib in ("pyqt4", "pyside"):
         msg_type, msg = args
-    elif qt_lib == "pyqt5":
+    elif qt_lib == ("pyqt5", "pyside2"):  # Is this correct for pyside2?
```
Comment (csachs) [inline vispy/app/backends/_qt.py]: Don't know more details, but `==` definetly doesn't work here.

`elif qt_lib in ("pyqt5", "pyside2"):`

Comment (davidh-ssec) [reply vispy/app/backends/_qt.py]: Wow, good catch. @hmaarrfk would you mind changing that?

**Comment by davidh-ssec** [comment]
@hmaarrfk If you don't mind, could you rebase your branch on to the current master and then force push to your branch to start the tests over again and see if everything passes (I committed some fixes to master today)?

**Comment by hmaarrfk** [comment]
@davidh-ssec, cool. I think i just rebased and pushed to my version. Travis should do his work now.

**Pull request merged**